src/lib/lmdb_storage/generic.ml

```diff
@@ -328,8 +328,183 @@ let%test_module "Lmdb storage tests" =
           cnt := !cnt + 1 ;
           `Continue ) ;
       assert (!cnt = !odd_cnt) ;
+
+      Deferred.unit
+
+    let%test_unit "Ro: Get and Iter operation within with_txn scope" =
+      let n = 300 in
+      let hm = Hashtbl.create (module Int) in
+      test_with_dir
+      @@ fun dir ->
+      let env, db = Rw.create dir in
+      Quickcheck.test
+        (Quickcheck.Generator.both uint32
+           (String.gen_with_length 100 Base_quickcheck.quickcheck_generator_char) )
+        ~trials:n
+        ~f:(fun (k, v_str) ->
+          let v = Bigstring.of_string v_str in
+          Hashtbl.add ~key:k ~data:v hm
+          |> function `Duplicate -> () | `Ok -> Rw.set ~env db k v ) ;
+
+      let ro_env, ro_db = Ro.create dir in
+      let hm_from_iter = Hashtbl.create (module Int) in
+      let (_ : Bigstring.t option) =
+        Ro.with_txn
+          ~f:(fun getter ->
+            getter.iter_ro
+              ~f:(fun key data_from_ro ->
+                let data_from_get =
+                  getter.get ro_db key
+                  |> Option.value_exn
+                       ~message:"cannot find element in get based on iter key"
+                in
+                [%test_eq: Bigstring.t] data_from_ro data_from_get ;
+                Hashtbl.add_exn ~key ~data:data_from_get hm_from_iter ;
+                `Continue )
+              ro_db ;
+            let first_key, _ =
+              Hashtbl.choose hm_from_iter
+              |> Option.value_exn ~message:"empty hash table"
+            in
+            getter.get ro_db first_key
+            |> Option.value_exn
+                 ~message:"cannot fetch first key from hm after iter" )
+          ro_env
+      in
+
+      [%test_eq: (int * Bigstring.t) List.t] (Hashtbl.to_alist hm)
+        (Hashtbl.to_alist hm_from_iter) ;
+      Deferred.unit
+
+    let%test_unit "Rw: Get, Set and Iter operation within with_txn scope" =
+      let n = 300 in
```

Comment:
This initialization section occurs repetitively in this test file, should be able to factor them out.